### PR TITLE
[DDW-712] Stake Pools - Search and Filter

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements Search and Filter for the Stake Pools list.

## Todo

- [x] Search pools at `Stake Pools` and `DelegationStepsChooseStakePoolDialog`
- [x] Filtering pools at `Stake Pools`
- [x] Clear button when searching, with "Clear" tooltip
- [x] Separator when the input has filtering options
- [x] Back To Top button when the Pools list is scrolled
- [x] List title with search info at `Stake Pools`
- [x] Back To Top to be on top of the Pool Tooltip
- [x] Add colors variables

## Screenshots
Screen Shot 2019-07-16 at 14 41 35
Screen Shot 2019-07-16 at 14 41 46
Screen Shot 2019-07-16 at 14 42 04
Screen Shot 2019-07-16 at 14 42 25

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1563299114388700)
- [ ] Test stake pools search at [`Staking > Stake Pools`](https://deploy-preview-1468--daedalus-storybook.netlify.com/?selectedKind=Staking&selectedStory=Stake%20Pools&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)
- [ ] Test stake pools search at [`Staking > DelegationStepsChooseStakePoolDialog`](https://deploy-preview-1468--daedalus-storybook.netlify.com/?selectedKind=Staking&selectedStory=DelegationStepsChooseStakePoolDialog&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by gnpf** [comment]
@daniloprates The button "Back to top" is not displayed on top of the tooltips

image

image

**Comment by a-rukin** [comment]
image
when I click text it goes on top
when I click bg it clicks stake poolfor info

**Comment by gnpf** [comment]
@daniloprates There is some overlapping between the clear button and the filters in the Japanese version. Also (maybe because of that), the clear button is not working.

image

File: source/renderer/app/components/widgets/BackToTopButton.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  font-size: 12px;
+  left: 50%;
+  line-height: 1.17;
+  opacity: 0.98;
+  padding: 6px 12px;
+  position: fixed;
+  text-align: center;
+  top: 144px;
+  transform: translateX(-50%);
+  opacity: 0;
```
Comment (tomothespian) [inline source/renderer/app/components/widgets/BackToTopButton.scss]: @daniloprates please order rules alphabetical

File: source/renderer/app/themes/daedalus/light-blue.js [hunk 2]
```diff
@@ -594,6 +599,9 @@ export const LIGHT_BLUE_THEME_CONFIG = {
     '--theme-staking-stake-pool-tooltip-text-color': '#5e6066',
     '--theme-staking-stake-pools-search-button-color': '#5e6066',
     '--theme-staking-stake-pools-search-icon-color': 'rgba(94, 96, 102, 1)',
+    '--theme-staking-stake-pools-search-clear-button-background-color':
```
Comment (tomothespian) [inline source/renderer/app/themes/daedalus/light-blue.js]: @daniloprates please include all newly created variables also in `utils/crateTheme.js`

**Comment by daniloprates** [comment]
> @daniloprates Great work. I leave few comments for you, please take a look

@tomothespian thanks for the great review. Filtering is actually not happening now, so I've removed it. All the other requests are done. Could you please re-check it?

**Pull request merged**